R/fit_trait_distributions.R

```diff
@@ -0,0 +1,130 @@
+#' Fit trait distributions
+#' @description Fits distributions to imputed traits and
+#' returns a tibble containing the fitted parameters.
+#' @param imputed_traits imputed trait and community data
+#' in long format
+#' @param distribution_type the type of statistical distribution to use.
+#' Character or named list. Currently accepts "normal","lognormal", and "beta".
+#' @note Distribution type should be a single character (e.g. "normal")
+#' or a named list (e.g. list(height="normal",mass="lognormal"))
+#' @description
+#'
+#' @return a tibble
+#'
+#' @importFrom stats var
+#' @importFrom e1071 skewness kurtosis
+#' @importFrom magrittr %>%
+#' @importFrom dplyr slice_sample group_by summarise
+#' summarize select group_by_at
+#' @importFrom purrr map_dfr
+#' @importFrom stats var
+#' @importFrom data.table :=
+#' @importFrom fitdistrplus fitdist
+
+#' @examples
+#' data(community)
+#' data(trait)
+#' imputed_traits <-trait_impute(comm = community, traits = trait,
+#'                  scale_hierarchy = c("Site", "PlotID"),
+#'                  taxon_col = "Taxon", value_col = "Value",
+#'                  trait_col = "Trait", abundance_col = "Cover")
+#' fitted_distributions <- fit_trait_distributions(
+#' imputed_traits = imputed_traits,
+#' distribution_type = "normal")
+#'
+#' @export
+fit_trait_distributions <- function(imputed_traits,
+                                    distribution_type = "normal") {
+
+  #Pull useful information from imputed traits object
+  value_col <- attributes(imputed_traits)$attrib$value_col
+  trait_col <- attributes(imputed_traits)$attrib$trait_col
+  taxon_col <- attributes(imputed_traits)$attrib$taxon_col
+  abundance_col <- attributes(imputed_traits)$attrib$abundance_col
+  scale_hierarchy <- attributes(imputed_traits)$attrib$scale_hierarchy
+
+  #If only a single type of distribution was supplied, apply it to all traits
+  if (length(distribution_type) == 1) {
+    distributions <- unique(as.data.frame(imputed_traits)[, trait_col])
+    dist_list <- replicate(n = length(distributions), expr = distribution_type)
+    names(dist_list) <- distributions
+    distribution_type <- as.list(dist_list)
+    rm(dist_list, distributions)
+  } #if statement
+
+  #Data checks
+
+  #check distribution type is supported
+  if (any(!unique(distribution_type) %in% c("beta", "normal", "lognormal"))) {
+    x <- distribution_type[!distribution_type %in%
+                            c("beta", "normal", "lognormal")]
+    stop(glue("distribution type '{x}' not supported "))
+  }
+
+
+  #Beta checks
+  if ("beta" %in% distribution_type) {
+
+    #make sure every species x hierarchy combination has at least 2 data points
+    beta_counts <- imputed_traits %>%
```

Comment:
inpute_traits has n_leaves column in output, so you don't need to recalculate n()

Reply:
Updated to use that column instead.